apm-agent-core/src/main/java/co/elastic/apm/agent/report/serialize/PayloadSerializer.java

```diff
@@ -71,4 +102,10 @@
     JsonWriter getJsonWriter();
 
     void writeBytes(byte[] bytes, int len);
+
+    class UninitializedException extends Exception {
```

Comment:
The main purpose of this type is to make anyone calling methods that throws it aware of the requirement for initialization, and the javadoc explains that. I prefer you notice this in compilation and for that it needs to be a checked Exception.

> it's not a recoverable

Not sure why - it may affect only the first send/serialization and not the subsequent ones.